### ISSUE TITLE
一个分享里面有多个文件，只会下其中一个文件

### ISSUE THREAD

**Issue by turygo** [banbanchs/pan-baidu-download#19]
比如http://pan.baidu.com/s/1c0cpoIW 这里有好几个文件分享，调用
pan download http://pan.baidu.com/s/1c0cpoIW 只会下载其中的一个文件

**Comment by banbanchs** [comment]
暂时不支持多文件分享的下，可以一个个点开下载

**Comment by serika00** [comment]
@banbanchs 问一问现在有计划做多文件下载及目录下载么？如果没有的话我打算尝试下做（不保证会做出来），如果有的话我就不重造了。

**Comment by banbanchs** [comment]
@serika00 暂时没有计划，欢迎PR

**Comment by serika00** [comment]
@banbanchs 结果做成了批量下载~

**Comment by serika00** [comment]
最近在下载文件夹里有多个文件情况的时候，当打包后整体比较大，1G左右，下载的时候就好容易会因为返回HTTP头的文件offset不正确导致下载中止，看来还是要弄个各文件分开单独下载的功能。

**Comment by banbanchs** [comment]
@serika00 以前观察的现象是是百度生成一个打包下载链接，同时服务器开始打包选中的文件，打包完成前下载链接为一个大小不确定的流，而且打包速度挺慢的。不过现在好像不同了，就是不太稳定

**Comment by Kagami** [comment]
Encountered same issue (if I understand your conversation with the help of google translate correctly).

File link: http://pan.baidu.com/s/1sjFSopf
Password: kvyq
Output:

```
$ pan download 'http://pan.baidu.com/s/1sjFSopf' --secret kvyq
(Live)[Girls' Generation] 140814 UMAX UHD M!Countdown in LA - Mr.Mr. + Mr.Taxi + Gee (HEVC 4K60fps 10bit).part1-batch.zip
[#1eb3f7 144KiB/4.0GiB(0%) CN:5 DL:70KiB ETA:16h40m31s]
04/19 14:20:24 [ERROR] CUID#10 - Download aborted. URI=http://www.baidupcs.com/rest/2.0/pcs/file?method=batchdownload&app_id=250528&zipcontent=%7B%22fs_id%22%3A%5B%2291386038216747%22%2C%22910789785161034%22%5D%7D&sign=DCb740ccc5511e5e8fedcff06b081203:WAJtZiAst94lne%2Bq%2FJ9ZvYIctSg%3D&uid=1141643709&time=1461151216&dp-logid=2546705301318186259&dp-callid=0
Exception: [AbstractCommand.cc:345] errorCode=8 URI=http://www.baidupcs.com/rest/2.0/pcs/file?method=batchdownload&app_id=250528&zipcontent=%7B%22fs_id%22%3A%5B%2291386038216747%22%2C%22910789785161034%22%5D%7D&sign=DCb740ccc5511e5e8fedcff06b081203:WAJtZiAst94lne%2Bq%2FJ9ZvYIctSg%3D&uid=1141643709&time=1461151216&dp-logid=2546705301318186259&dp-callid=0
  -> [HttpResponse.cc:86] errorCode=8 Invalid range header. Request: 1090519040-2179989503/4359412700, Response: 1090519040-1107296255/4359412700
[#1eb3f7 256KiB/4.0GiB(0%) CN:4 DL:58KiB ETA:20h21m39s]
04/19 14:20:27 [ERROR] CUID#12 - Download aborted. URI=http://www.baidupcs.com/rest/2.0/pcs/file?method=batchdownload&app_id=250528&zipcontent=%7B%22fs_id%22%3A%5B%2291386038216747%22%2C%22910789785161034%22%5D%7D&sign=DCb740ccc5511e5e8fedcff06b081203:WAJtZiAst94lne%2Bq%2FJ9ZvYIctSg%3D&uid=1141643709&time=1461151216&dp-logid=2546705301318186259&dp-callid=0
Exception: [AbstractCommand.cc:345] errorCode=8 URI=http://www.baidupcs.com/rest/2.0/pcs/file?method=batchdownload&app_id=250528&zipcontent=%7B%22fs_id%22%3A%5B%2291386038216747%22%2C%22910789785161034%22%5D%7D&sign=DCb740ccc5511e5e8fedcff06b081203:WAJtZiAst94lne%2Bq%2FJ9ZvYIctSg%3D&uid=1141643709&time=1461151216&dp-logid=2546705301318186259&dp-callid=0
  -> [HttpResponse.cc:86] errorCode=8 Invalid range header. Request: 545259520-2179989503/4359412700, Response: 545259520-562036735/4359412700
[#1eb3f7 256KiB/4.0GiB(0%) CN:3 DL:42KiB ETA:28h5m5s]
04/19 14:20:29 [ERROR] CUID#9 - Download aborted. URI=http://www.baidupcs.com/rest/2.0/pcs/file?method=batchdownload&app_id=250528&zipcontent=%7B%22fs_id%22%3A%5B%2291386038216747%22%2C%22910789785161034%22%5D%7D&sign=DCb740ccc5511e5e8fedcff06b081203:WAJtZiAst94lne%2Bq%2FJ9ZvYIctSg%3D&uid=1141643709&time=1461151216&dp-logid=2546705301318186259&dp-callid=0
Exception: [AbstractCommand.cc:345] errorCode=8 URI=http://www.baidupcs.com/rest/2.0/pcs/file?method=batchdownload&app_id=250528&zipcontent=%7B%22fs_id%22%3A%5B%2291386038216747%22%2C%22910789785161034%22%5D%7D&sign=DCb740ccc5511e5e8fedcff06b081203:WAJtZiAst94lne%2Bq%2FJ9ZvYIctSg%3D&uid=1141643709&time=1461151216&dp-logid=2546705301318186259&dp-callid=0
  -> [HttpResponse.cc:86] errorCode=8 Invalid range header. Request: 2179989504-3270508543/4359412700, Response: 2179989504-2196766719/4359412700
[#1eb3f7 1.3MiB/4.0GiB(0%) CN:2 DL:106KiB ETA:11h6m42s]
04/19 14:20:36 [ERROR] CUID#11 - Download aborted. URI=http://www.baidupcs.com/rest/2.0/pcs/file?method=batchdownload&app_id=250528&zipcontent=%7B%22fs_id%22%3A%5B%2291386038216747%22%2C%22910789785161034%22%5D%7D&sign=DCb740ccc5511e5e8fedcff06b081203:WAJtZiAst94lne%2Bq%2FJ9ZvYIctSg%3D&uid=1141643709&time=1461151216&dp-logid=2546705301318186259&dp-callid=0
Exception: [AbstractCommand.cc:345] errorCode=8 URI=http://www.baidupcs.com/rest/2.0/pcs/file?method=batchdownload&app_id=250528&zipcontent=%7B%22fs_id%22%3A%5B%2291386038216747%22%2C%22910789785161034%22%5D%7D&sign=DCb740ccc5511e5e8fedcff06b081203:WAJtZiAst94lne%2Bq%2FJ9ZvYIctSg%3D&uid=1141643709&time=1461151216&dp-logid=2546705301318186259&dp-callid=0
  -> [HttpResponse.cc:86] errorCode=8 Invalid range header. Request: 3270508544-4359412699/4359412700, Response: 3270508544-3287285759/4359412700
 *** Download Progress Summary as of Tue Apr 19 14:21:17 2016 ***
```

**Comment by banbanchs** [comment]
@Kagami Confirmed. This is baidu's problem. I will implement new batch download function in the future.

As for now, I suggest you can try [this](https://github.com/acgotaku/BaiduExporter), **or** save the files to your own account and create a private share link for each files.

I am sorry for the inconvenience. :(

**Comment by brainstorm** [comment]
Your script worked quite well on the first run, looking forward to that improved batch/ranges download function since BaiduExporter seems to require windows:

https://github.com/acgotaku/BaiduExporter/tree/master/aria2c

:_/

**Comment by Kagami** [comment]
It doesn't require Windows, see [here](http://www.cnx-software.com/2015/01/14/downloading-files-on-baidu-or-via-http-bittorrent-or-metalink-in-linux-with-baiduexporter-aria2-and-yaaw/) guide for Linux.

**Comment by Kagami** [comment]
I think this is fixed by #33 and need to be closed.
I tried latest commit with the example above and everything seems to be working. Great job!

**Comment by banbanchs** [comment]
@Kagami Thanks for testing. @serika00 :+1:
